gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java

```diff
@@ -140,6 +142,62 @@ public void eagerInitialization_fails_withInvalidCredentialsConfiguration() {
             "non-existent.json (The system cannot find the file specified)");
   }
 
+  @Test
+  public void read_file_throws_exception() throws IOException {
+    URI bucketPath = new Path("gs://test-file").toUri();
+    GoogleHadoopFileSystem myGhfs = new GoogleHadoopFileSystem();
+    myGhfs.initialize(bucketPath, new Configuration());
+
+    // custom channel
+    SeekableByteChannel channel =
+        new SeekableByteChannel() {
+          @Override
+          public boolean isOpen() {
+            return false;
+          }
+
+          @Override
+          public void close() throws IOException {
+            throw new IOException("close not implemented");
+          }
+
+          @Override
+          public int read(ByteBuffer buffer) throws IOException {
+            throw new IOException("read throws exception");
```

Comment:
nit: capture message in a variable and compare the cause while asserting.

Reply:
PTAL